shopfloor/migrations/13.0.1.2.0/post-migration.py

```diff
@@ -0,0 +1,61 @@
+# Copyright 2020 Camptocamp SA (http://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import json
+import logging
+
+from odoo import SUPERUSER_ID, api
+
+_logger = logging.getLogger("shopfloor." + __name__)
+
+
+def _compute_logs_new_values(env):
+    log_entries = env["shopfloor.log"].search([])
+    for entry in log_entries:
+        new_vals = {}
+        for fname in ("params", "headers", "result"):
+            if not entry[fname]:
+                continue
+            # make it json-like
+            replace_map = [
+                ("{'", '{"'),
+                ("'}", '"}'),
+                ("':", '":'),
+                (": '", ': "'),
+                ("',", '",'),
+                (", '", ', "'),
+                ("False", "false"),
+                ("True", "true"),
+                ("None", "null"),
+                ("\\xa0", " "),
+            ]
+            json_val = entry[fname]
+            for to_replace, replace_with in replace_map:
+                json_val = json_val.replace(to_replace, replace_with)
+            try:
+                val = json.loads(json_val)
+            except Exception:
+                # fail gracefully and do not break the whole thing
+                # just for not being able to convert a value.
+                # We don't use these values as json yet, no harm.
+                _logger.warning(
+                    "`%s` JSON convert failed for record %d", (fname, entry.id)
+                )
+            new_vals[fname] = json.dumps(val, indent=4, sort_keys=True)
```

Comment:
`val` won't be defined if an exception occurs, the `json.dumps` has to be in a `else:` clause no?

Reply:
good catch! I thought about fixing it and then I forgot :sweat:

Reply:
fixed!